### PR TITLE
Book: rewrite intro, overview, portability; add guide-start

### PR DESCRIPTION
**Pull request by dhardy**
@vks would you be happy to review?

### CODE REVIEW

File: src/crates.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+### Deterministic generators
+
+The following crates implement [PRNGs]:
+
+-   [`rand_chacha`] provides generators using the ChaCha cipher
+-   [`rand_hc`] implements a generator using the HC-128 cipher
+-   [`rand_isaac`] implements the ISAAC generators
+-   [`rand_pcg`] implements a small selection of PCG generators
+-   [`rand_xoshiro`] implements the SplitMix and Xoshiro generators
+-   [`rand_xorshift`] implements the basic Xorshift generator
```
Comment (vks) [inline src/crates.md]: Maybe subdivide this into into generators that are cryptographically secure and generators that aren't.

Is it worth it to mention the `rand_isaac` and `rand_xorshift` generators? I consider them deprecated.

File: src/features.md [hunk 2]
```diff
@@ -4,42 +4,63 @@ Some functionality is gated behind Cargo features, which must be enabled as
...
+Almost all Rand libraries can be used without Rust's standard library:
 
-The `nightly` feature enables `simd_support`, along with some additional
-SIMD functionality not available on stable compilers.
+-   For the [`rand`] and [`rand_chacha`] crates, one must disable default features
+    (`default-features = false`) and some functionality is lost.
+-   The [`rand_core`] and [`rand_jitter`] crates assume `no_std` by default, but
+    support some additional functionality when the `std` feature is enabled.
+-   [`getrandom`] (and by extension [`rand_os`]) has limited support for `no_std`.
+    It does not use of feature flags but detects based on the target platform.
```
Comment (vks) [inline src/features.md]: This sentence is not clear to me. Maybe replace "of" with "any"? What does it detect based on the target platform?

File: src/guide-start.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    println!("Lets go in this direction: {}", arrows_iter.choose(&mut rng).unwrap());
+    let mut nums = [1, 2, 3, 4, 5];
+    nums.shuffle(&mut rng);
+    println!("I shuffled my {:?}", nums);
+}
+```
+
+The first thing you may have noticed is that we imported everything from the
+[prelude]. This is
+the lazy way to `use` rand, and like the
```
Comment (vks) [inline src/guide-start.md]: Those two lines can be joined.

File: src/guide-start.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+remember to import the [`Rng`] trait!
+
+The Rand library automatically initialises a secure, thread-local generator
+on demand. This can be accessed via the [`thread_rng`] and [`random`] functions.
+For more on this topic, see [Random generators](guide-gen.md).
+
+While the [`random`] function can only sample values in a [`Standard`]
+(type-dependent) manner, [`thread_rng`] gives you a handle to a generator.
+All generators implement the [`Rng`] trait, which provides the [`gen`] and
+[`gen_range`] methods used above.
```
Comment (vks) [inline src/guide-start.md]: I would mention `sample` here for choosing the distribution.

File: src/crates.md (hunk 1 above)
Comment (dhardy) [reply src/crates.md]: The point is just to make a quick overview of crates, not go into the RNGs.

**Comment by dhardy** [comment]
Updated.

Most of the guide and the updating guide still needs to be done.

File: src/crates.md (hunk 1 above)
Comment (vks) [reply src/crates.md]: Fair enough, in this case the list should indeed be exhaustive.

File: src/guide-start.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        println!("y is: {}", y);
+        println!("Number from 0 to 9: {}", rng.gen_range(0, 10));
+    }
+    
+    // We can also interact with iterators and slices:
+    let arrows_iter = "➡⬈⬆⬉⬅⬋⬇⬊".chars();
+    println!("Lets go in this direction: {}", arrows_iter.choose(&mut rng).unwrap());
+    let mut nums = [1, 2, 3, 4, 5];
+    nums.shuffle(&mut rng);
+    println!("I shuffled my {:?}", nums);
```
Comment (vks) [inline src/guide-start.md]: The sentence printed to the screen is a bit weird:
```
I shuffled my [3, 5, 2, 1, 4]
```
Maybe just add "numbers:"?

Comment (dhardy) [reply src/guide-start.md]: It's not a very good one, I admit. Hint: 'shuffled' is a synonym of 'confused'.

Comment (vks) [reply src/guide-start.md]: Fair enough.

**Pull request merged**